presto-hive/src/main/java/com/facebook/presto/hive/metastore/thrift/ThriftHiveMetastore.java

```diff
@@ -746,7 +763,43 @@ public void alterTable(String databaseName, String tableName, Table table)
     }
 
     @Override
-    public void addPartitions(String databaseName, String tableName, List<Partition> partitions)
+    public void addPartitions(String databaseName, String tableName, List<PartitionWithStatistics> partitionsWithStatistics)
+    {
+        List<Partition> partitions = partitionsWithStatistics.stream()
+                .map(ThriftMetastoreUtil::toMetastoreApiPartition)
+                .collect(toImmutableList());
+        addMetastorePartitions(databaseName, tableName, partitions);
+        try {
+            if (partitionsWithStatistics.stream().allMatch(p -> p.getStatistics().getColumnStatistics().isEmpty())) {
+                return;
+            }
+            Table table = getTable(databaseName, tableName)
+                    .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
```

Comment:
In my opinion, it is bad API design on hive metastore to have those methods that take partition names. All of them should really be taking partVals. But anyways, that ship sailed.

When using this API, care needs to be taken. I don't think it's acceptable to make all these extra `getTable` calls. 

* Based on my brief reading, the pre-existing `getPartitionStatistics` don't need to turn partVals into partition names, it should do all the conversions the other way around.
* For the new `addPartitions` and `alterPartition`, I'm thinking maybe the right way to approach this is to use include the partition column names in the thrift-API-side (as opposed to the extended-side) `PartitionWithStatistics`. Otherwise, all these calls are  going to turn into a significant number of `getTable` calls.

Reply:
> Based on my brief reading, the pre-existing getPartitionStatistics don't need to turn partVals into partition names, it should do all the conversions the other way around.

`getPartitionStatistics` accepts a list of partition names. It needs `Table` to get the column names. Column names are needed by the HiveMetastore api to get statistcs. 

> For the new addPartitions and alterPartition, I'm thinking maybe the right way to approach this is to use include the partition column names in the thrift-API-side (as opposed to the extended-side) PartitionWithStatistics. Otherwise, all these calls are going to turn into a significant number of getTable calls.

I'm going to add `partitionName` field to the `PartitionWithStatistics` object